File: Domain/Movies/Movie.cs
using Domain.Users;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Movies
{
    public class Movie
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string PosterPath { get; set; }
        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
        public DateTime? DateWatched { get; set; } = null;
        public MovieUser User { get; set; }
        public MovieRating Rating { get; set; }
        public MovieComment Comment { get; set; }
        public List<string> DefaultCategories { get; set; }
        public ICollection<UserCategory>? UserCategories { get; set; }

    }
}

File: MovieDiary/ClientApp/src/models/Movie.ts
export type Movie = {
  id?: string;
  title: string;
  description: string;
  dateCreated: Date;
  dateWatched: string;
  posterPath: string;
  user: Profile;
  rating: Rating;
  comment: Comment;
  // categories: Category[] | string[];
  defaultCategories: string[];
  userCategories: Category[];
};

export type SearchedMovie = {
  poster_path: string;
  title: string;
  release_date: string;
  // highlight: boolean;
  vote_average: number;
  overview: string;
  backdrop_path: string;
  genre_ids: number[];
};

export interface UserAdjustedSearchedMovie extends SearchedMovie {
  rating: { value: number };
  comment: { text: string };
  userCategories: Category[] | string[]; //pri vyberu kategorii string[], pak se preklopi do Category[]
  defaultCategories: string[];
  dateWatched: any;
}

export type Profile = {
  id?: string;
  name?: string;
};

export type Rating = {
  //id: string;
  value: number;
};
export type Comment = {
  id?: string;
  text: string;
};

export type Category = {
  id?: string;
  name: string;
};
